### PR TITLE
Children transactions + Gift Cards / Incognito

### PR DESCRIPTION
**Pull request by znarf**
Screenshot 2021-09-16 at 10 38 07

### CODE REVIEW

File: components/CreateCollectiveMiniForm.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ const msg = defineMessages({
   },
   examples: {
     id: 'examples',
-    defaultMessage: 'e.g. {examples}',
+    defaultMessage: 'e.g., {examples}',
```
Comment (znarf) [inline components/CreateCollectiveMiniForm.js:103]: I expect this to be solved in another PR but this is currently broken on `main`.

File: components/transactions/TransactionItem.js [hunk 2]
```diff
@@ -135,19 +152,25 @@ const TransactionItem = ({ displayActions, collective, transaction, onMutationSu
...
               </Container>
               <P mt="4px" fontSize="12px" lineHeight="20px" color="black.700" data-cy="transaction-details">
-                {isCredit ? (
-                  <FormattedMessage
-                    id="Transaction.from"
-                    defaultMessage="from {name}"
-                    values={{ name: <StyledLink as={LinkCollective} collective={fromAccount} /> }}
-                  />
-                ) : (
+                {i18nTransactionType(intl, transaction.type)}
```
Comment (znarf) [inline components/transactions/TransactionItem.js:155]: Add Transaction.type

File: components/transactions/TransactionItem.js [hunk 3]
```diff
@@ -135,19 +152,25 @@ const TransactionItem = ({ displayActions, collective, transaction, onMutationSu
...
+                  <Fragment>
+                    <FormattedMessage
+                      id="Transaction.from"
+                      defaultMessage="from {name}"
+                      values={{ name: <StyledLink as={LinkCollective} collective={fromAccount} /> }}
+                    />
+                    &nbsp;
+                  </Fragment>
+                }
+                {
```
Comment (znarf) [inline components/transactions/TransactionItem.js:167]: From and To

File: components/transactions/TransactionItem.js [hunk 4]
```diff
@@ -212,8 +235,11 @@ const TransactionItem = ({ displayActions, collective, transaction, onMutationSu
             {hasExpense && <ExpenseStatusTag status={expense.status} fontSize="9px" px="6px" py="2px" />}
           </Flex>
         </Flex>
-        {hasOrder && (
+        {hasOrder && [CONTRIBUTION, ADDED_FUNDS].includes(transaction.kind) && (
```
Comment (znarf) [inline components/transactions/TransactionItem.js:238]: Restrict to CONTRIBUTION and ADDED_FUNDS

File: components/transactions/TransactionItem.js [hunk 5]
```diff
@@ -246,6 +272,11 @@ const TransactionItem = ({ displayActions, collective, transaction, onMutationSu
             <ExpenseTags expense={expense} />
           </Container>
         )}
+        {!hasExpense && (!hasOrder || ![CONTRIBUTION, ADDED_FUNDS].includes(transaction.kind)) && (
```
Comment (znarf) [inline components/transactions/TransactionItem.js:275]: If not Expense or CONTRIBUTION / ADDED_FUNDS with order.

File: pages/transactions.js [hunk 6]
```diff
@@ -268,6 +303,52 @@ class TransactionsPage extends React.Component {
...
+                  label={
+                    <FormattedMessage
+                      id="transactions.excludeIncognito"
+                      defaultMessage="Exclude Incognito transactions"
+                    />
+                  }
+                />
+              )}
+            </Flex>
+
```
Comment (znarf) [inline pages/transactions.js:351]: New Checkboxes for filtering

File: pages/transactions.js [hunk 7]
```diff
@@ -114,6 +119,9 @@ const getVariablesFromQuery = query => {
     dateTo,
     searchTerm: query.searchTerm,
     kind: query.kind ? parseTransactionKinds(query.kind) : getDefaultKinds(),
+    includeIncognitoTransactions: !query.ignoreIncognitoTransactions,
+    includeGiftCardsTransactions: !query.ignoreGiftCardsTransactions,
+    includeChildrenTransactions: !query.ignoreChildrenTransactions,
```
Comment (znarf) [inline pages/transactions.js:124]: Sync GraphQL query.

File: lib/i18n/transaction.js [hunk 8]
```diff
@@ -60,7 +60,7 @@ const i18nKind = defineMessages({
   },
   [TransactionKind.PREPAID_PAYMENT_METHOD]: {
     id: 'Transaction.kind.PREPAID_PAYMENT_METHOD',
-    defaultMessage: 'Prepaid',
+    defaultMessage: 'Prepaid payment method',
```
Comment (znarf) [inline lib/i18n/transaction.js:63]: More precise name, previous was confusing.

File: components/transactions/TransactionsDownloadCSV.js [hunk 9]
```diff
@@ -188,10 +188,15 @@ const TransactionsDownloadCSV = ({ collective, client, query }) => {
...
-    url.searchParams.set('includeIncognitoTransactions', '1');
+    if (!query.ignoreGiftCardTransactions) {
+      url.searchParams.set('includeGiftCardTransactions', '1');
+    }
+    if (!query.ignoreIncognitoTransactions) {
+      url.searchParams.set('includeIncognitoTransactions', '1');
+    }
+    if (!query.ignoreChildrenTransactions) {
+      url.searchParams.set('includeChildrenTransactions', '1');
+    }
```
Comment (znarf) [inline components/transactions/TransactionsDownloadCSV.js:199]: Support more parameters in CSV.

File: pages/transactions.js [hunk 10]
```diff
@@ -158,6 +168,30 @@ class TransactionsPage extends React.Component {
...
+      this.setState({ hasGiftCards });
+    }
+
+    const hasIncognito =
+      (this.props.data?.transactions?.nodes || []).some(el => el.fromAccount?.isIncognito) ||
+      this.props.query.ignoreIncognitoTransactions;
+    if (isNil(this.state.hasIncognito) && hasIncognito) {
+      this.setState({ hasIncognito });
+    }
+
```
Comment (znarf) [inline pages/transactions.js:194]: Detect if we want to display the new checkboxes. Most of the time not.

File: components/transactions/TransactionItem.js [hunk 11]
```diff
@@ -305,6 +336,7 @@ TransactionItem.propTypes = {
     id: PropTypes.string,
     uuid: PropTypes.string,
     type: PropTypes.string,
+    kind: PropTypes.string,
```
Comment (SudharakaP) [inline components/transactions/TransactionItem.js:339]: Maybe a slightly stricter enforcement of the `PropTypes` is;

```suggestion
    kind: PropTypes.oneOf(Object.values(TransactionKind))
```

File: components/collective-page/sections/Budget.js [hunk 12]
```diff
@@ -36,11 +36,13 @@ export const budgetSectionWithHostQuery = gqlV2/* GraphQL */ `
...
       hasExpense: false
       kind: $kind
+      includeIncognitoTransactions: true
       includeGiftCardTransactions: true
+      includeChildrenTransactions: true
     ) {
       ...TransactionsQueryCollectionFragment
     }
-    expenses(account: { slug: $slug }, limit: $limit) {
+    expenses(account: { slug: $slug }, limit: $limit, includeChildrenExpenses: true) {
```
Comment (kewitz) [inline components/collective-page/sections/Budget.js:45]: Not sure if we should include children's transactions on the collective page by default.

File: pages/transactions.js [hunk 13]
```diff
@@ -158,6 +173,30 @@ class TransactionsPage extends React.Component {
...
+    if (isNil(this.state.hasGiftCards) && hasGiftCards) {
+      this.setState({ hasGiftCards });
+    }
+
+    const hasIncognito =
+      (this.props.data?.transactions?.nodes || []).some(el => el.account?.isIncognito) ||
+      this.props.query.ignoreIncognitoTransactions;
+    if (isNil(this.state.hasIncognito) && hasIncognito) {
+      this.setState({ hasIncognito });
+    }
```
Comment (kewitz) [inline pages/transactions.js:198]: Ideally these properties would come from the backend, so we don't end up adding the checkbox as we paginate through transactions.

Comment (znarf) [reply pages/transactions.js:198]: True but good enough for now as backend cost/complexity would be much bigger.

Comment (znarf) [reply pages/transactions.js:198]: I'm okay having the checkbox appear while paginating.

File: components/collective-page/sections/Budget.js (hunk 12 above)
Comment (znarf) [reply components/collective-page/sections/Budget.js:45]: The approach is to have it by default everywhere for consistency. Unless there is a huge blocker, let's see how it goes.
